uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
+    var vm = Object.create(this),
+      dayTypes = [],
+      leaveRequests = [],
+      publicHolidays = [],
+      leaveRequestStatuses = [];
+
+    vm.absencePeriods = [];
+    vm.absenceTypes = [];
+    vm.calendar = {};
+    vm.loading = false;
+    vm.months = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+    //Select current month as default
+    vm.selectedMonths = [vm.months[new Date().getMonth()]];
+    vm.selectedPeriod = null;
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    vm.isPublicHoliday = function (date) {
+      return !!publicHolidays[new Date(date).getTime()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    vm.getDayName = function (date) {
+      return new Date(date).toString().substr(0,3);
+    };
+
+    /**
+     * Returns the calendar information for a specific month
+     *
+     * @param  {int} month
+     * @return {array}
+     */
+    vm.getMonthData = function (month) {
+      if (vm.calendar.days) {
+        var iterator,
+          dates = Object.keys(vm.calendar.days),
+          date,
+          length = dates.length,
+          datesForTheMonth = [];
+
+        for (iterator = 0; iterator < length; iterator++) {
+          date = new Date(parseInt(dates[iterator]));
+          if (date.getMonth() === month) {
+            datesForTheMonth.push(vm.calendar.days[dates[iterator]]);
+          }
+        }
+
+        return datesForTheMonth;
+      }
+    };
+
+    /**
+     * Refresh all leave request and calendar data
+     */
+    vm.refresh = function () {
+      vm.loading = true;
+      loadLeaveRequest()
+        .then(function () {
+          vm.loading = false;
+        });
+    };
+
+    (function init() {
+      vm.loading = true;
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadPublicHolidays(),
+        loadStatuses(),
+        loadDayTypes()
+      ]).then(function () {
+          vm.legendCollapsed = false;
+          return loadLeaveRequest();
+        })
+        .then(function () {
+          vm.loading = false;
+        });
+    })();
+
+    /**
+     * Returns the leave request which is in range of the sent date
+     *
+     * @param  {string} date
+     * @return {object}
+     */
+    function getLeaveRequest(date) {
+      var iterator,
+        length = leaveRequests.length,
+        dates;
+
+      for (iterator = 0; iterator < length; iterator++) {
```

Comment:
More expressive and terser functional programming code (not 100% sure is correct, but something like this)
```js
function getLeaveRequest(date) {
  return _.find(leaveRequests, function (leaveRequest) {
    return leaveRequest.dates.some(function (date2) { // give better parm name
      return moment(date2).isSame(date);
    });
  });
});